apis/spec/v1alpha3/cluster_workload_resource_mapping.go

```diff
@@ -0,0 +1,117 @@
+// resource. For each object matching the path the name, env and volumeMounts expressions are resolved to find those
+// structures.
+type ClusterWorkloadResourceMappingContainer struct {
+	// Path is the JSONPath within the workload resource that matches an existing fragment that is container-like.
+	Path string `json:"path"`
+	// Name is a Restricted JSONPath that references the name of the container with the container-like workload resource
+	// fragment. If not defined, container name filtering is ignored.
+	// +optional
+	Name string `json:"name,omitempty"`
+	// Env is a Restricted JSONPath that references the slice of environment variables for the container with the
+	// container-like workload resource fragment. The referenced location is created if it does not exist. Defaults
+	// to `.envs`.
+	// +optional
+	Env string `json:"env,omitempty"`
+	// VolumeMounts is a Restricted JSONPath that references the slice of volume mounts for the container with the
+	// container-like workload resource fragment. The referenced location is created if it does not exist. Defaults
+	// to `.volumeMounts`.
+	// +optional
+	VolumeMounts string `json:"volumeMounts,omitempty"`
+}
+
+// ClusterWorkloadResourceMappingSpec defines the desired state of ClusterWorkloadResourceMapping
+type ClusterWorkloadResourceMappingSpec struct {
+	// Versions is the collection of versions for a given resource, with mappings.
+	Versions []ClusterWorkloadResourceMappingTemplate `json:"versions,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster
+// +kubebuilder:storageversion
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+
+// ClusterWorkloadResourceMapping is the Schema for the clusterworkloadresourcemappings API
+type ClusterWorkloadResourceMapping struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ClusterWorkloadResourceMappingSpec `json:"spec,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// ClusterWorkloadResourceMappingList contains a list of ClusterWorkloadResourceMapping
+type ClusterWorkloadResourceMappingList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []ClusterWorkloadResourceMapping `json:"items"`
+}
+
+var defaultTemplate = ClusterWorkloadResourceMappingTemplate{
+	Version:     "*",
+	Annotations: "",
+	Volumes:     ".spec.template.spec.volumes",
+	Containers: []ClusterWorkloadResourceMappingContainer{
+		{
+			Path:         ".spec.template.spec.containers[*]",
+			Name:         ".name",
+			Env:          ".env",
+			VolumeMounts: ".volumeMounts",
+		},
+		{
+			Path:         ".spec.template.spec.initContainers[*]",
+			Name:         ".name",
+			Env:          ".env",
+			VolumeMounts: ".volumeMounts",
+		},
+	},
+}
+
+func DefaultTemplate() ClusterWorkloadResourceMappingTemplate {
```

Comment:
Changed.

Long-term, I'd like to use a [`webhook.Defaulter`](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.2/pkg/webhook/admission#Defaulter), since it'll let us reuse some code from when we write validation admission webhooks.  However, since there's only one place we use this in the code, changing that isn't going to be difficult.